spec/models/grade_entry_form_spec.rb

```diff
@@ -0,0 +1,241 @@
+      @grade_entry_form = make_grade_entry_form_with_multiple_grade_entry_items
+      @grade_entry_items = @grade_entry_form.grade_entry_items
+    end
+
+    it 'verify the correct value is returned when the student has grades for some of the questions' do
+      student = Student.make
+      grade_entry_student_with_some_grades = @grade_entry_form.grade_entry_students.find_by(user: student)
+      grade_entry_student_with_some_grades.grades.create(grade_entry_item: @grade_entry_items[0], grade: 3)
+      grade_entry_student_with_some_grades.grades.create(grade_entry_item: @grade_entry_items[1], grade: 7)
+      expect(grade_entry_student_with_some_grades.all_blank_grades?).to be false
+    end
+
+    it 'verify the correct value is returned when the student has grades for all of the questions' do
+      student = Student.make
+      grade_entry_student_with_some_grades = @grade_entry_form.grade_entry_students.find_by(user: student)
+      grade_entry_student_with_some_grades.grades.create(grade_entry_item: @grade_entry_items[0], grade: 3)
+      grade_entry_student_with_some_grades.grades.create(grade_entry_item: @grade_entry_items[1], grade: 7)
+      grade_entry_student_with_some_grades.grades.create(grade_entry_item: @grade_entry_items[2], grade: 8)
+      expect(grade_entry_student_with_some_grades.all_blank_grades?).to be false
+    end
+
+    it 'verify the correct value is returned when the student has grades for none of the questions' do
+      student1 = Student.make
+      grade_entry_student_with_no_grades = @grade_entry_form.grade_entry_students.find_by(user: student1)
+      expect(grade_entry_student_with_no_grades.all_blank_grades?).to be true
+    end
+  end
+
+  # Tests for calculate_released_average
+  context 'Calculate the average of the released marks: ' do
+    before(:each) do
+      @grade_entry_form = make_grade_entry_form_with_multiple_grade_entry_items
+      @grade_entry_form_none_released = make_grade_entry_form_with_multiple_grade_entry_items
+      @grade_entry_items = @grade_entry_form.grade_entry_items
+      # Set up 5 GradeEntryStudents
+      (0..4).each do |i|
+        student = Student.make
+        grade_entry_student = @grade_entry_form.grade_entry_students.find_by(user: student)
+        # Give the student a grade for all three questions for the grade entry form
+        (0..2).each do |j|
+          grade_entry_student.grades.create(grade_entry_item: @grade_entry_items[j],
+                                            grade: 5 + i + j)
+        end
+
+        # The marks will be released for 3 out of the 5 students
+        if i <= 2
+          grade_entry_student.released_to_student = true
+        else
+          grade_entry_student.released_to_student = false
+        end
+        grade_entry_student.save
+      end
+    end
+
+    it 'verify the correct value is returned when there are no marks released' do
+      expect(@grade_entry_form_none_released.calculate_released_average()).to eq 0
```

Comment:
Omit the `()` for method calls that have no parameters. (This occurs in other places as well.)